### DISCUSSION TITLE
Packaging for Debian

### DISCUSSION THREAD

**Discussion by winnieXY** [g4ixt/QtTinySA#31] [General]
Hello,
i would like to provide QtTinySA as debian package within the official debian package pool for every interested user within the scope of the debian-hams project.

I hope that your are fine with that :-) If users find issues within QtTinySA I would forward the information to you, so that you don't have to cope with the bugtracking system of debian. 

Currently I've some minor issues packaging the project (mainly related to legal issues):

- Would it be possible to provide within the source code (e.g. within the individual python files) a statement that this python file is copyright by you (+ maybe others) and the license you used. Due to the LICENSE file within the repository I assume that the complete project is GPL-3  (or GPL-3+ ?). 

This is (for example) a statement from gr-satellites (a extension to gnuradio):

```
# Copyright 2016 Daniel Estevez <daniel@destevez.net>
#
# This file is part of gr-satellites
#
# SPDX-License-Identifier: GPL-3.0-or-later
#
```
Something like this would be perfectly sufficient.

Furthermore I need a e-mailadresse how to contact you as upstream author of the package. 

You can see the current state of packaging here:
https://salsa.debian.org/winnieXY/qttinysa

and the ITP here: 
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1063230

**Comment by g4ixt** [comment]
Hello,

Yes that is fine with me.  I presume that you will take the code from the main branch at some point each time you release an updated package?  It might be helpful to tell me before that so if I have some update plans from my development branch I can take it into account.

Do I need to include an eMail address, or would this change be sufficient?  Or do you need all the 'acknowledgement' lines to be comments starting '#'?

!/usr/bin/env python3
 -*- coding: utf-8 -*-

Copyright 2024 Ian Jefferson G4IXT
SPDX-License-Identifier: GPL-3.0-or-later

"""TinySA Ultra GUI programme using Qt5 and PyQt.

This code attempts to replicate some of the TinySA Ultra on-screen commands and to provide PC control.
Development took place on Kubuntu 22.04LTS with Python 3.9 and PyQt5 using Spyder in Anaconda.

TinySA and TinySA Ultra are trademarks of Erik Kaashoek and are used with permission.

TinySA commands are based on Erik's Python examples: http://athome.kaashoek.com/tinySA/python/

Serial communication commands are based on Martin's Python NanoVNA/TinySA Toolset:
https://github.com/Ho-Ro

"""

**Reply by winnieXY** [reply]
That's perfectly sufficient :-) If you don't want to have a public e-mailadress within the internet (which is something I can totally understand) we'll find another solution  - usually there is somewhere a emailadress available, but I guess this discussion forum here will do, too - I'll ask.

**Reply by winnieXY** [reply]
For the initial upload I took the sourcecode of the tagged release (0.9.1) and my plan was to look out for new tagged releases and upload them. So you can release whenever you want. In debian there is a tool (called uscan) which will notify me as soon as there is a new tag available.

**Comment by g4ixt** [comment]
If you can send me a private message I will send my eMail address to you. I don't know how to do that on here but I know private messages are possible on Erik's TinySA group in groups.io

**Comment by g4ixt** [comment]
I have incorporated the personalised config database code into v0.10.0 which is now in 'development' branch.  The code you committed didn't seem to work properly so I have amended it - I hope it still does what you expected.  I removed the 'read only' code because I could see no benefit to it.
